internal/handler/handler.go

```diff
@@ -258,13 +258,19 @@ func (h *Handler) cleanupCappedCollection(ctx context.Context, db backends.Datab
 
 	h.L.Debug("cleanupCappedCollection: stats before", zap.Any("stats", statsBefore))
 
-	if statsBefore.SizeCollection < cInfo.CappedSize && statsBefore.CountDocuments < cInfo.CappedDocuments {
```

Comment:
When  `cInfo.CappedDocuments == 0`, `statsBefore.CountDocuments < cInfo.CappedDocuments` always results in false, triggering unnecessary cleanup.

Reply:
Nice catch!